lib/ReactViews/Tool.tsx

```diff
@@ -44,4 +47,48 @@ class Tool extends React.Component<PropsType> {
   }
 }
 
+interface ToolButtonProps extends ToolProps {
+  icon: { id: string };
+}
+
+@observer
+export class ToolButton extends React.Component<ToolButtonProps> {
+  @computed
+  get isThisToolOpen() {
+    const currentTool = this.props.viewState.currentTool;
+    return (
+      currentTool &&
+      currentTool.getToolComponent === this.props.getToolComponent
+    );
+  }
+
+  toggleOpen() {
+    const { viewState } = this.props;
+    if (this.isThisToolOpen) {
+      viewState.closeTool();
+    } else {
+      viewState.openTool({ ...this.props, showCloseButton: false });
```

Comment:
You spread props here but should it be stored on `params`? (or even a toolProps key) Haven't looked at your place editor implementation but at least we don't pollute the `Tool` interface with props

Reply:
The only `ToolButton` specific prop here is icon. The rest are meant for the `Tool` component which will correctly pass down the `params` object as props to the user component. I have improved it a bit by making it more explicit what we pass down.